src/test/java/com/volasoftware/tinder/services/implementations/VerificationTokenServiceImplTest.java

```diff
@@ -0,0 +1,145 @@
+        verificationToken.setExpiresAt(LocalDateTime.now().minusDays(1));
+
+        //when
+        when(repository.findByToken(uuidToken)).thenReturn(Optional.of(verificationToken));
+
+        //then
+        VerificationTokenExpiredException exception = assertThrows(
+                VerificationTokenExpiredException.class, () -> {
+                    service.verifyToken(uuidToken);
+                });
+
+        assertEquals(SecurityConstant.TOKEN_EXPIRED, exception.getMessage());
+    }
+
+    @Test
+    void testVerifyTokenWhenEmailAlreadyVerifiedThenThrowEmailAlreadyVerifiedException() {
+        //given
+        Account account = generateAccount();
+        VerificationToken verificationToken = generateVerificationToken(account);
+        String uuidToken = verificationToken.getToken();
+        verificationToken.setVerifiedAt(LocalDateTime.now().plusMinutes(5));
+
+        //when
+        when(repository.findByToken(uuidToken)).thenReturn(Optional.of(verificationToken));
+
+        //then
+        EmailAlreadyVerifiedException exception = assertThrows(
+                EmailAlreadyVerifiedException.class, () -> {
+                    service.verifyToken(uuidToken);
+                });
+
+        assertEquals(MailConstant.ALREADY_CONFIRMED, exception.getMessage());
+    }
+
+    @Test
+    void testVerifyTokenWhenTokenDoesNotExistThenThrowTokenNotExistsException() {
+        //given
+        String token = "tokenDoesNotExist";
+
+        //when
+        when(repository.findByToken(token)).thenReturn(Optional.empty());
+
+        //then
+        VerificationTokenExpiredException exception = assertThrows(
+                VerificationTokenExpiredException.class, () -> {
+                    service.verifyToken(token);
+                });
+
+        assertEquals(SecurityConstant.TOKEN_EXPIRED, exception.getMessage());
+    }
+
+    @Test
+    void testVerifyTokenWhenTokenExistThenReturnTheAccountDto() {
+        //given
+        Account account = generateAccount();
+        VerificationToken verificationToken = generateVerificationToken(account);
+
+        //when
+        when(repository.findByToken(verificationToken.getToken())).thenReturn(Optional.of(verificationToken));
+        when(repository.save(verificationToken)).thenReturn(verificationToken);
+
+        Account expectedAccount = service.verifyToken(verificationToken.getToken());
+
+        //then
+        verify(repository, times(1)).save(verificationToken);
+
+        assertNotNull(expectedAccount);
+        assertEquals(account.getEmail(), expectedAccount.getEmail());
+    }
+
+    private VerificationToken generateVerificationToken(
+            Account account) {
+
+        VerificationToken verificationToken = new VerificationToken();
+        String randomToken = UUID.randomUUID().toString();
+        verificationToken.setToken(randomToken);
+        verificationToken.setAccount(account);
+        verificationToken.setCreatedDate(LocalDateTime.now());
```

Comment:
In tests, it's not good idea to use LocalDateTime.now(), its hard to be tested, just create concrete one instead (ex. LocalDateTime.of(...))